### PR TITLE
Feat/defaults

### PR DESCRIPTION
**Pull request by RWOverdijk**

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -300,6 +300,36 @@ describe('Homefront', () => {
     });
   });
 
+  describe('.defaults()', () => {
+    it("Should set the defaults (left to right) provided", () => {
+      let homefront = new Homefront({foo: {bar: {bat: 'value', baz: 'value', deep: {also: 'works'}}}});
```
Comment (doktordirk) [inline test/index.test.js]: would be easier to read as multiline

File: index.js [hunk 2]
```diff
@@ -149,6 +149,18 @@ class Homefront {
   }
 
   /**
+   * Method allowing you to set missing keys (backwards-applied defaults) nested.
+   *
+   * @param {String|Array} key
+   * @param {*}            defaults
+   *
+   * @returns {Homefront}
```
Comment (doktordirk) [inline index.js]: that.s a new instance or this?

File: README.md [hunk 3]
```diff
@@ -51,6 +51,13 @@ homefront.merge(
 // And, sorry, the same again, but static.
 Homefront.merge({foo: 'bar'}, {bat: 'baz'});
 
+// And sort of the same, apply defaults.
```
Comment (doktordirk) [inline README.md]: could be clearer. 'post-apply missing defaults' or so

File: index.js (hunk 2 above)
Comment (RWOverdijk) [inline index.js]: That's how you type "this" in doc blocks.

**Pull request merged**